src/header.jsx

```diff
@@ -115,3 +119,47 @@ const ViewSelector = ({ isGrid, setIsGrid, sortBy, setSortBy }) => {
         </Select>
     );
 };
+
+const UploadButton = ({ currentDir, files }) => {
+    const ref = useRef();
+
+    const handleClick = () => {
+        ref.current.click();
+    };
+
+    const onUpload = e => {
+        const uploadedFile = e.target.files[0];
+
+        let fileName = uploadedFile.name;
+
+        // eslint-disable-next-line no-loop-func
+        while (files.some(f => f.name === fileName)) {
+            fileName += " (1)";
+        }
+
+        const reader = new FileReader();
+        reader.readAsArrayBuffer(uploadedFile, "UTF-8");
+        reader.onload = readerEvent => {
+            const channel = cockpit.channel({
+                binary: true,
+                payload: "fsreplace1",
+                path: currentDir + fileName
+            });
+
+            const buffer = readerEvent.target.result;
+            channel.send(buffer);
```

Comment:
Likewise, you can't send the complete file all at once. For medium-sized (a few MB) this will block the bridge and thus the UI for too long, and for large (GB) files you'll just create an out-of-memory crash.